File: lib/src/main/java/com/sha/formvalidator/model/Options.java
package com.sha.formvalidator.model;

import android.content.Context;
import android.text.TextUtils;

import com.sha.formvalidator.R;

public class Options {
    public String errorMessage;
    public boolean required = true;
    public ValidationType validationType;
    public String customValidationType;

    public String regex;
    public String dateFormat;

    public String emptyErrorMessage;

    public int minNumber;
    public int maxNumber;

    public float floatMinNumber;
    public float floatMaxNumber;

    public String emptyErrorMessage(Context context){
        return !TextUtils.isEmpty(emptyErrorMessage) ?
                emptyErrorMessage :
                context.getString(R.string.required);
    }
}

File: lib/src/main/java/com/sha/formvalidator/model/Options.kt
package com.sha.formvalidator.model

import android.content.Context
import android.text.TextUtils

import com.sha.formvalidator.R

class Options {
    var errorMessage: String = ""
    var required = true
    var validationType: ValidationType? = null
    var customValidationType: String = ""

    var regex: String = ""
    var dateFormat: String = ""

    var emptyErrorMessage: String = ""

    var minNumber: Int = 0
    var maxNumber: Int = 0

    var floatMinNumber: Float = 0.toFloat()
    var floatMaxNumber: Float = 0.toFloat()

    fun emptyErrorMessage(context: Context): String {
        return if (!TextUtils.isEmpty(emptyErrorMessage))
            emptyErrorMessage
        else
            context.getString(R.string.required)
    }
}
